src/test/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilderUnitTest.java

```diff
@@ -316,6 +319,49 @@ public void testIndexingOnTheFly() {
         }
     }
 
+  @Test
+  public void testIndexingOnTheFlyForPath() throws IOException {
+    final VariantContextWriterBuilder builder = new VariantContextWriterBuilder()
+        .setReferenceDictionary(dictionary)
+        .setOption(Options.INDEX_ON_THE_FLY);
+
+    try (FileSystem fs = Jimfs.newFileSystem("test", Configuration.unix())) {
+      Path vcfPath = fs.getPath("testIndexingOnTheFlyForPath" + IOUtil.VCF_FILE_EXTENSION);
+      Path vcfIdxPath = Tribble.indexPath(vcfPath);
+      VariantContextWriter writer = builder.setOutputPath(vcfPath).build();
+      writer.close();
+      Assert.assertTrue(Files.exists(vcfIdxPath),
+          String.format("VCF index not created for %s / %s", vcfPath, vcfIdxPath));
+
+      if (bcfIdx.exists())
```

Comment:
It may not be necessary because `.setOutputFile` now converts to a `Path` internally, but just to be sure I've now changed the code to use `.setOutputPath` so we explicitly pass a `Path` to the writer. We still check using files but that's outside of the code under test.